### ISSUE TITLE
[BUG] DHF 5.1.0: Total count does not match failed + successful events

### ISSUE THREAD

**Issue by josvanroosmalen** [marklogic/marklogic-data-hub#3670]
It seems that DHF5.1.0 is repeating early DHF 4 issues: total Events does not match succesfull events + failed events. 

For our "W" POC (@grtjn ) sourcing "K" data we are using DHF5 + Pipes. We create a URI based on Pipes String template (@sashamitrovich ). 

Then I run the harmonization using the following task (batchsize = 1 is for debug):

```
task wHarmonizeK(type: com.marklogic.gradle.task.RunFlowTask, group: project.name) {
  flowName="wK"
  batchSize=Integer.parseInt(project.property("wHarmonizationBatchSize"))
  threadCount=Integer.parseInt(project.property("wHarmonizationThreadCount"))
  showOptions=true
  failHard=true
  batchSize=1
  steps=["2"]
  jobId="wHarmonizeK-123"
}
```

My Pipes graph is slicing the wrong data, causing errors in URI. However I would expect:

- 1 error doc for each error input doc. That is not the case. There are 0 error docs so I need to "guess" which docs go wrong. 
- In the job overview the failed Events = 0, what is clearly not the case.  The input is 100 docs, OK = 84 so I expect 16 to be reported failure. 

```
{
"job": {
"jobId": "wHarmonizeK-123", 
"flow": "wK", 
"user": "admin", 
"lastAttemptedStep": "2", 
"lastCompletedStep": "2", 
"jobStatus": "finished", 
"timeStarted": "2020-03-09T20:31:46.2871864+01:00", 
"timeEnded": "2020-03-09T20:31:58.0973486+01:00", 
"stepResponses": {
"2": {
"flowName": "wK", 
"stepName": "Harmonizewk", 
"stepDefinitionName": "Harmonizewk", 
"stepDefinitionType": "custom", 
"stepOutput": [
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75910243210001\\nAKR1.12432298\\nAKR1.12432298\\nAKR1...\") -- Invalid URI format: \"/wK/75910243210001&#10;AKR1.12432298&#10;AKR1.12432298&#10;AKR1.100000008287163&#10;109781764&#10;197938462&#10;20070402006042&#10;AKR1.15404966&#10;AKR1.14625221&#10;AKR1.15051358&#10;AKR1.100000001740633&#10;AKR1.100000006618068&#10;75910243210001&#10;AKR1.12432298&#10;1003886844&#10;107329001&#10;109781764&#10;18050101008225&#10;1000142203&#10;20191031004543&#10;500001453937.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75910243210002\\nAKR1.13259802\\nAKR1.13259802\\nAKR1...\") -- Invalid URI format: \"/wK/75910243210002&#10;AKR1.13259802&#10;AKR1.13259802&#10;AKR1.100000005996778&#10;109781764&#10;197938462&#10;20070402006042&#10;AKR1.15404966&#10;AKR1.14625221&#10;AKR1.15051358&#10;AKR1.100000001266901&#10;AKR1.100000004839712&#10;75910243210002&#10;AKR1.13259802&#10;1003886845&#10;107329001&#10;109781764&#10;18050101008225&#10;1000142203&#10;20191031004543&#10;500001453937.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75910309610048\\nAKR1.12948044\\nAKR1.12948044\\nAKR1...\") -- Invalid URI format: \"/wK/75910309610048&#10;AKR1.12948044&#10;AKR1.12948044&#10;AKR1.100000005997064&#10;109856864&#10;325103755&#10;20031230003227&#10;AKR1.13931234&#10;AKR1.100000001266901&#10;AKR1.100000004839738&#10;75910309610048&#10;AKR1.12948044&#10;1003834522&#10;558043923&#10;109856864&#10;20031230003227&#10;AKR1.13931234&#10;20191031002967&#10;500001432683.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75910309610189\\nAKR1.12761403\\nAKR1.12761403\\n1098...\") -- Invalid URI format: \"/wK/75910309610189&#10;AKR1.12761403&#10;AKR1.12761403&#10;109856864&#10;325103755&#10;20031230003227&#10;AKR1.13931234&#10;75910309610189&#10;AKR1.12761403&#10;1003834523&#10;558043923&#10;109856864&#10;20031230003227&#10;AKR1.13931234&#10;20191031002967&#10;500001432683.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75910309610151\\nAKR1.12736522\\nAKR1.12736522\\n1098...\") -- Invalid URI format: \"/wK/75910309610151&#10;AKR1.12736522&#10;AKR1.12736522&#10;109856864&#10;325103755&#10;20031230003227&#10;AKR1.13931234&#10;75910309610151&#10;AKR1.12736522&#10;1003834524&#10;558043923&#10;109856864&#10;20031230003227&#10;AKR1.13931234&#10;20191031002967&#10;500001432683.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75910467570000\\nAKR1.100000005851960\\nAKR1.1000000...\") -- Invalid URI format: \"/wK/75910467570000&#10;AKR1.100000005851960&#10;AKR1.100000005851960&#10;325103755&#10;AKR1.100000001423165&#10;AKR1.100000005579236.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/76060471970000\\n1000832873\\n1003570817\\nKRS3.76060...\") -- Invalid URI format: \"/wK/76060471970000&#10;1000832873&#10;1003570817&#10;KRS3.76060471970000&#10;KRS5.76060471970000&#10;109793020&#10;AKR1.14501242&#10;1001813099&#10;KRS1.76060471970000&#10;KRS1.76060471970000.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75970491410013\\nAKR1.12836571\\nAKR1.12836571\\n1086...\") -- Invalid URI format: \"/wK/75970491410013&#10;AKR1.12836571&#10;AKR1.12836571&#10;108674505&#10;109798286&#10;18020203011089&#10;AKR1.15421381&#10;20050905005857&#10;AKR1.15350738&#10;75970491410013&#10;AKR1.12836571&#10;1003844713&#10;107672354&#10;109798286&#10;18020203011089&#10;AKR1.15421381&#10;20191031002937&#10;500001432661.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/75970480510001\\nAKR1.12443670\\nAKR1.12443670\\nAKR1...\") -- Invalid URI format: \"/wK/75970480510001&#10;AKR1.12443670&#10;AKR1.12443670&#10;AKR1.100000006180784&#10;108674505&#10;109784905&#10;20050905005857&#10;AKR1.15350738&#10;AKR1.13925388&#10;AKR1.14241139&#10;AKR1.100000001281345&#10;AKR1.100000004910974&#10;75970480510001&#10;AKR1.12443670&#10;1003844712&#10;107672354&#10;109784905&#10;18050313013527&#10;1000185967&#10;20191031002937&#10;500001432661.json\"", 
"com.marklogic.client.FailedRequestException: Local message: failed to apply resource at resources/ml:runFlow: Bad Request. Server Message: XDMP-URI: xdmp.documentGetCollections(\"/wK/76090625270000\\nAKR1.12850359\\n1001436529\\n4541377...\") -- Invalid URI format: \"/wK/76090625270000&#10;AKR1.12850359&#10;1001436529&#10;454137791&#10;20190219001580&#10;500000461683.json\""
], 
"fullOutput": null, 
"status": "completed step 2", 
"totalEvents": 100, 
"successfulEvents": 84, 
"failedEvents": 0, 
"successfulBatches": 84, 
"failedBatches": 0, 
"success": true, 
"stepStartTime": "2020-03-09T20:31:46.6312213+01:00", 
"stepEndTime": "2020-03-09T20:31:57.3627931+01:00"
}
}
}
}```

**Comment by ryanjdew** [comment]
This appears to be a duplicate of https://github.com/marklogic/marklogic-data-hub/issues/2895. @josvanroosmalen Can you confirm that this issue is resolved for you in 5.2?

**Comment by rjrudin** [comment]
DHFPROD-6522 also fixed an issue related to incorrect counts. I can't verify that every count issue is fixed now, but @josvanroosmalen  - if this comes up again with DHF 5.4 or higher, please open a JIRA bug for it.
